codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/Waiters.java

```diff
@@ -692,4 +732,50 @@ private String generateRetryableName(
         waiterName = StringUtils.uncapitalize(waiterName);
         return String.format("%sStateRetryable", waiterName);
     }
+
+
+    /**
+     * Returns the MemberShape wrt to the provided Shape and name.
+     * For eg, If shape `A` has MemberShape `B`, and the name provided is `B` as string.
+     * We return the MemberShape `B`.
+     *
+     * @param model the generation model.
+     * @param shape the shape that is walked to retreive the shape matching provided name.
+     * @param name  name is a single scope path string, and should only match to one or less shapes.
+     * @return MemberShape matching the name.
+     */
+    private MemberShape getComparedMember(Model model, Shape shape, String name) {
+
+        name = name.replaceAll("\\[\\]", "");
+
+        // if shape is a simple shape, just return shape as member shape
+        if (shape instanceof SimpleShape) {
+            return shape.asMemberShape().get();
+        }
+
+        switch (shape.getType()) {
+            case STRUCTURE:
+                StructureShape st = shape.asStructureShape().get();
+                for (MemberShape memberShape : st.getAllMembers().values()) {
+                    if (name.equalsIgnoreCase(memberShape.getMemberName())) {
+                        return memberShape;
+                    }
+                }
+                break;
+
+            case LIST:
+                ListShape listShape = shape.asListShape().get();
+                MemberShape listMember = listShape.getMember();
+                Shape listTarget = model.expectShape(listMember.getTarget());
+                return getComparedMember(model, listTarget, name);
+
+            default:
+                // TODO: add support for * usage with jmespath expression.
+                return null;
```

Comment:
Would be good to have tracking item for this.

Reply:
Tracking item paired with https://github.com/aws/smithy-go/issues/238